src/me/john000708/barrels/Barrels.java

```diff
@@ -232,29 +273,35 @@ public boolean onRightClick(ItemUseEvent itemUseEvent, Player player, ItemStack
             public boolean onRightClick(ItemUseEvent itemUseEvent, Player player, ItemStack itemStack) {
                 if (!SlimefunManager.isItemSimiliar(itemStack, ID_CARD, false)) return false;
                 Block clickedBlock = itemUseEvent.getClickedBlock();
-                ItemStack idCard = itemStack;
-                ItemMeta meta = idCard.getItemMeta();
-                List<String> lore = idCard.getItemMeta().getLore();
+                // No need to reference itemStack again in a new variable.
+                //ItemStack idCard = itemStack;
+                ItemMeta meta = itemStack.getItemMeta();
+                List<String> lore = itemStack.getItemMeta().getLore();
 
                 if (lore.get(0).equals("")) {
                     lore.set(0, ChatColor.translateAlternateColorCodes('&', "&0" + player.getUniqueId().toString()));
                     lore.set(1, ChatColor.translateAlternateColorCodes('&', "&fBound to: " + player.getName()));
                     meta.setLore(lore);
-                    idCard.setItemMeta(meta);
+                    itemStack.setItemMeta(meta);
                     player.sendMessage(ChatColor.GREEN + "ID Card bound.");
                 } else if (clickedBlock != null && BlockStorage.hasBlockInfo(clickedBlock) && BlockStorage.checkID(clickedBlock).startsWith("BARREL_") && BlockStorage.getLocationInfo(clickedBlock.getLocation(), "whitelist") != null && BlockStorage.getLocationInfo(clickedBlock.getLocation(), "owner").equals(player.getUniqueId().toString())) {
                     String whitelistedPlayers = BlockStorage.getLocationInfo(clickedBlock.getLocation(), "whitelist");
                     if (!whitelistedPlayers.contains(ChatColor.stripColor(lore.get(0)))) {
                         BlockStorage.addBlockInfo(clickedBlock, "whitelist", whitelistedPlayers + ChatColor.stripColor(lore.get(0)) + ";");
-                        player.getInventory().setItem(player.getInventory().getHeldItemSlot(), InvUtils.decreaseItem(itemStack, 1));
+                        // Fixes issue #6.
+                        //player.getInventory().setItem(player.getInventory().getHeldItemSlot(), InvUtils.decreaseItem(itemStack, 1));
+                        int amount = itemStack.getAmount();
+                        if (amount <= 1) {
+                            itemStack.setAmount(0);
+                        }
+                        else itemStack.setAmount(amount - 1);
                         player.sendMessage(ChatColor.GREEN + "Player successfully whitelisted!");
                     } 
                     else {
                         player.sendMessage(ChatColor.RED + "The player is already whitelisted.");
                     }
-
-                }
-                return false;
+                } else return false;
```

Comment:
Don't return false in that case.
Just remove this.

The method has a return false; statement at the very top, If the statement at the top isn't triggered, then this method should always return true.